### ISSUE TITLE
Refactor can't deal with UnicodeSyntax

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/hlint#465]
From https://github.com/ndmitchell/hlint/issues/223#issuecomment-375145897:

hm, it doesn't seem to work when I enable the `-XUnicodeSyntax` extension.
my example file:
```haskell
module Model.File where

import qualified Data.Bson as Bson
import Data.Int (Int64)
import Data.Time.Clock (UTCTime)
import GHC.Generics (Generic)

-- | Represents a file stored in Mongo GridFS.
data File = File
  { _id        ∷ Bson.ObjectId
  , length     ∷ Int64
  , uploadDate ∷ UTCTime
  , fileName   ∷ String
  } deriving (Eq, Show, Generic)
```

```bash
 ▲ haskell/2018/spreadsheet-api cat .hlint.yaml
- arguments: [-XUnicodeSyntax]

 ▲ haskell/2018/spreadsheet-api hlint src/Model/File.hs --refactor
refactor: (RealSrcSpan SrcSpanOneLine "src/Model/File.hs" 10 16 17,"parse error on input \8216\8759\8217")
CallStack (from HasCallStack):
  error, called at src/Main.hs:189:22 in main:Main

 △ haskell/2018/spreadsheet-api hlint --version
HLint v2.0.11, (C) Neil Mitchell 2006-2017
```

Reported by @vyorkin

@ndmitchell says: Seems like the .yaml file is a red herring - my guess is refactor can't deal with unicode, perhaps because I should be passing more flags to it.

**Comment by ndmitchell** [comment]
Covered by #504.
